pkg/virtctl/utils/utils.go

```diff
@@ -12,13 +12,16 @@ import (
 // AttachConsole attaches stdin and stdout to the console
 // in -> stdinWriter | stdinReader -> console
 // out <- stdoutReader | stdoutWriter <- console
-func AttachConsole(stdinReader, stdoutReader *io.PipeReader, stdinWriter, stdoutWriter *io.PipeWriter, message string, resChan <-chan error) error {
+func AttachConsole(stdinReader, stdoutReader *io.PipeReader, stdinWriter, stdoutWriter *io.PipeWriter, message string, resChan <-chan error) (err error) {
 	stopChan := make(chan struct{}, 1)
 	writeStop := make(chan error)
 	readStop := make(chan error)
-	state, err := terminal.MakeRaw(int(os.Stdin.Fd()))
-	if err != nil {
-		return fmt.Errorf("Make raw terminal failed: %s", err)
+	var state *terminal.State
+	if terminal.IsTerminal(int(os.Stdin.Fd())) {
+		state, err = terminal.MakeRaw(int(os.Stdin.Fd()))
+		if err != nil {
+			return fmt.Errorf("Make raw terminal failed: %s", err)
+		}
```

Comment:
nit: maybe better to defer restoring of the terminal right here instead of doing it at the end of the function
```go
defer terminal.Restore(int(os.Stdin.Fd()), state)
```

Reply:
Fixed. Makes a lot of sense. :+1:

Reply:
And it may actually resolve a few unpleasant situations which I had sometimes when interrupting virtcl at a bad moment.